chapter-06/1_load/bookworm_internal_test.go

```diff
@@ -39,23 +39,22 @@ func TestLoadBookworms_Success(t *testing.T) {
 	for name, testCase := range tests {
 		t.Run(name, func(t *testing.T) {
 			got, err := loadBookworms(testCase.bookwormsFile)
-			if err != nil && !testCase.wantErr {
-				t.Fatalf("expected an error %s, got an empty one", err.Error())
-			}
 
-			if err == nil && testCase.wantErr {
+			switch {
+			case err != nil && !testCase.wantErr:
+				t.Fatalf("expected an error %s, got an empty one", err.Error())
+			case err == nil && testCase.wantErr:
 				t.Fatalf("expected no error, got one %s", err.Error())
```

Comment:
we are missing the case where we expect an error and get one

Reply:
I found it makes the switch hard to read, so I refactored. The alternative was:
```
switch {
	case testCase.wantErr && err == nil:
		t.Fatalf("expected no error, got one %s", err.Error())
	case err != nil && !testCase.wantErr:
		t.Fatalf("expected an error %s, got an empty one", err.Error())
	case !testCase.wantErr && !equalBookworms(t, got, testCase.want):
		t.Fatalf("different result: got %v, expected %v", got, testCase.want)
}
```

Reply:
Actually the yews forest is no better:
```
if !testCase.wantErr {
	if err != nil {
		t.Fatalf("expected no error, got one %s", err.Error())
	}

	if !equalBookworms(t, got, testCase.want) {
		t.Fatalf("different result: got %v, expected %v", got, testCase.want)
	}
} else if err == nil {
	t.Fatalf("expected an error, got none")
}
```

Reply:
```go
if testCase.wantErr  {
  if err == nil {
    t.Fatal("expected err, got nothing")
  }
  return
}

// we aren't expecting errors here, this should be the happy path
if err != nil {
  t.Fatalf("expected no error, got %v", err)
}
if !equalBookworms( got, testCase.want) {
  t.Fatalf("different result: got %v, expected %v", got, testCase.want)
}
```